test/unit/deadline_job_attachments/test_path_mapping.py

```diff
@@ -0,0 +1,337 @@
+
+    # transform passes through other paths
+    assert applier.transform("/other/path") == "/other/path"
+
+    # strict_transform raises for other paths
+    with pytest.raises(ValueError):
+        applier.strict_transform("/other/path")
+
+
+def test_source_windows_rule(dest_paths: DestPaths):
+    applier = _PathMappingRuleApplier(
+        [
+            PathMappingRule(PathFormat.WINDOWS.value, "C:\\Shared", str(dest_paths.path1)),
+            PathMappingRule(PathFormat.WINDOWS.value, "C:\\proJects", str(dest_paths.path2)),
+            PathMappingRule(PathFormat.WINDOWS.value, "D:\\tmp", str(dest_paths.path3)),
+        ]
+    )
+
+    # All three rules can be used with both regular and strict transform
+    assert applier.transform("C:\\Shared") == dest_paths.path1
+    assert applier.transform("C:\\proJects") == dest_paths.path2
+    assert applier.transform("D:\\tmp") == dest_paths.path3
+    assert applier.strict_transform("C:\\Shared") == dest_paths.path1
+    assert applier.strict_transform("C:\\proJects") == dest_paths.path2
+    assert applier.strict_transform("D:\\tmp") == dest_paths.path3
+
+    # Windows is case insensitive but case preserving
+    assert applier.transform("C:\\ShArEd") == dest_paths.path1
+    assert (
+        applier.transform("C:\\PROJECTS\\Case\\Of\\tail\\PreServed")
+        == dest_paths.path2 / "Case" / "Of" / "tail" / "PreServed"
+    )
+
+    # transform passes through other paths
+    assert applier.transform("C:\\other\\path") == "C:\\other\\path"
+
+    # strict_transform raises for other paths
+    with pytest.raises(ValueError):
+        applier.strict_transform("C:\\other\\path")
+
+
+def test_source_posix_rule_edge_cases(dest_paths: DestPaths):
+    applier = _PathMappingRuleApplier(
+        [
+            PathMappingRule(PathFormat.POSIX.value, "/mnt/shared", str(dest_paths.path1)),
+            PathMappingRule(PathFormat.POSIX.value, "/mnt/shared/projects", str(dest_paths.path2)),
+            PathMappingRule(PathFormat.POSIX.value, "/tmp", str(dest_paths.path3)),
+        ]
+    )
+
+    # Paths that are not transformed
+    assert applier.transform("") == ""
+    assert applier.transform("/") == "/"
+    assert applier.transform("/other/path") == "/other/path"
+    assert applier.transform("/mnt/other/path") == "/mnt/other/path"
+    assert applier.transform("/Mnt/shared") == "/Mnt/shared"
+
+    # Edge cases with unicode and spaces
+    assert applier.transform("/mnt/shared/файл.txt") == dest_paths.path1 / "файл.txt"
+    assert (
+        applier.transform("/mnt/shared/file with spaces.txt")
+        == dest_paths.path1 / "file with spaces.txt"
+    )
+
+    # The second rule never applies, because the first rule is shorter and is used instead
+    assert applier.transform("/mnt/shared/projects") == dest_paths.path1 / "projects"
+    assert (
+        applier.transform("/mnt/shared/projects/file.txt")
+        == dest_paths.path1 / "projects" / "file.txt"
+    )
```

Comment:
Seems like the more specific / longer rule should take precedence?

Reply:
I've updated the implementation to do this, and added a link to the openjd-specifications wiki in the docstring.